packages/headless/src/controllers/core/context/headless-core-internal-context.ts

```diff
@@ -44,10 +49,16 @@ export function buildInternalCoreContext(engine: CoreEngine): InternalContext {
   }
   const {dispatch} = engine;
   return {
-    add(contextKey: string, contextValue: ContextValue) {
+    add(contextKey: ReservedContextKey, contextValue: ContextValue) {
+      if (!isReservedContextKey(contextKey)) {
+        throw new UnreservedContextKeyError(contextKey);
+      }
       dispatch(addContext({contextKey, contextValue}));
     },
-    remove(key: string) {
+    remove(contextKey: ReservedContextKey) {
+      if (!isReservedContextKey(contextKey)) {
```

Comment:
Does it really matter if a non-reserved context key-value gets removed through this method?

Reply:
Mh I like the idea of having a strict contingency, for all operations.
This increases the likeliness of keys to be properly reserved :)